app/grants/clr.py

```diff
@@ -587,3 +615,144 @@ def predict_clr_live(grant, contributor, amount, is_postive_vote=True):
+    check_token = lambda token_address: len(token_address) == 42
+    check_contract = lambda token_address, abi : w3.eth.contract(token_address, abi=abi)
+    check_event_transfer =  lambda contract_address, search, txid : w3.eth.filter({ "address": contract_address, "topics": [search, txid]})
+    get_decimals = lambda contract : int(contract.functions.decimals().call())
+
+    # Colors for Console.
+    class bcolors:
+        HEADER = '\033[95m'
+        OKBLUE = '\033[94m'
+        OKGREEN = '\033[92m'
+        WARNING = '\033[93m'
+        FAIL = '\033[91m'
+        ENDC = '\033[0m'
+        BOLD = '\033[1m'
+        UNDERLINE = '\033[4m'
+
+    # scrapper settings
+    ethurl = "https://etherscan.io/tx/"
+    user_agent = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/35.0.1916.47 Safari/537.36'
+    headers = {'User-Agent': user_agent}
+
+    # scrapes etherscan to get the replaced tx
+    def getReplacedTX(tx):
+        sleep(2)  # 2s delay to avoid getting the finger from etherscan
+        response = requests.get(ethurl + tx, headers=headers)
+        soup = BeautifulSoup(response.content, "html.parser")
+        # look for span that contains the dropped&replaced msg
+        p = soup.find("span", "u-label u-label--sm u-label--warning rounded")
+        if not p:
+            return 'dropped'
+        if "Replaced" in p.text:  # check if it's a replaced tx
+            # get the id for the replaced tx
+            q = soup.find(href=re.compile("/tx/0x"))
+            return q.text
+        else:
+            return "dropped"
+
+    def transaction_eth(transaction_tax):
+        transaction_receipt = w3.eth.getTransactionReceipt(transaction_tax)
+        if (transaction_receipt != None and transaction_receipt.cumulativeGasUsed >= 2100):
+            transaction_hash = transaction_receipt.transactionHash.hex()
+            transaction = check_transaction(transaction_hash)
+            if transaction.value > 0.001:
+                amount = w3.fromWei(transaction.value, 'ether')
+                print(
+                    f"{bcolors.OKGREEN} {index_element} txid: {transaction_tax[:10]} {amount} ETH -> status: 1 {bcolors.ENDC}")
+            else:
+                print(
+                    f"{bcolors.FAIL}{bcolors.UNDERLINE} {index_element} txid: {transaction_tax[:10]} -> status: 0 - amount was off by 0.001 {bcolors.ENDC}")
+
+    def transaction_status(transaction, txid):
+        """This function is core for check grants transaction list"""
+        contract_address = transaction.to
+        contract = check_contract(contract_address, erc20_abi)
+        approve_event = check_event_transfer(contract.address, SEARCH_METHOD_APPROVAL, txid)
+        transfer_event = check_event_transfer(contract.address, SEARCH_METHOD_TRANSFER, txid)
+        deposit_event = check_event_transfer(contract.address, SEARCH_METHOD_DEPOSIT, txid)
```

Comment:
`bool(deposit_event)` seems to always be true no matter what `deposit_event.get_all_entries()` returns

are you sure this is actually measuring token transfers

Reply:
for example this txn seems to show as though its had transfers but it has not 

https://etherscan.io/tx/0x05c45f8bc90b0d4da6f2c289b62b0641933690e5ed5b0cae1808cbc070c1c17e

Reply:
I will check this event response

Reply:
I have isolated the answer from the check_event_transfe () function now giving the correct result.

Reply:
what was the change? i dont see any changes to `check_event_transfer()`

Reply:
I think deposit event can be discarded, only two transactions are made after contributing to a grant, #1 approval and #2 transfer.

Also the SEARCH_METHOD_APPROVAL variable should be '0x8c5be1e5ebec7d5bd14f71427d1e84f3dd0314c0f7b2291e5b200ac8c7c3b925'.

Reply:
can you share a script you're using to generate the hashes? there doesnt seem to be a kecak in 4.6.0 so im having a hard time creating my own hashes

the only method i really need workin gis the transfer one, as thats the one that tells the grants system that the contribution is the real deal

Reply:
hash = web3.sha3(text="Approval(address,address,uint256)").hex()